### DISCUSSION TITLE
Truncate long project directories on the repo Web home page

### DISCUSSION THREAD

**Discussion by mcandre** [community/community#86537] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

When a project has a rather long list of files in the top level directory, then it becomes a chore to scroll all the way down to the README directions.

I would love for the default GitHub repo page generator to automatically truncate long lists of files, e.g. to just ten files, and then provide an ellipsis link to re-expand the file list.

I think GitHub already does this on mobile browsers. It should do so on desktop as well.

No, I don't want to have to *convince* people to nest very large projects into deeper subfolder hierarchies. For one thing, that often breaks the build process.

**Comment by STREGA** [comment]
It would be nice if the truncation prioritized culling hidden files.
I doubt many people are eager to review `.gitignore`, `.gitattributes`, or `.github` unless they are learning.
